### PR TITLE
Rule label-has-for is deprecated

### PR DESCRIPTION
**Pull request by spyl94**
A deprecated rule should not be present in recommended or strict configurations.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -73,7 +73,6 @@ module.exports = {
             ],
           },
         ],
-        'jsx-a11y/label-has-for': 'error',
         'jsx-a11y/label-has-associated-control': 'error',
```
Comment (ljharb) [inline src/index.js]: hmm, the addition of the associated control rule to these configs should have been semver-major

**Comment by ljharb** [comment]
Either way, “deprecated” is a documentation thing to notify people; i don’t see much point in removing the rule from these configs now (as opposed to later, when the actual rule is removed)

**Comment by spyl94** [comment]
Changelog (https://github.com/evcohen/eslint-plugin-jsx-a11y/releases/tag/v6.1.0) explains that this rule has been replaced by `label-has-associated-control` which is already present.

Right now I need to disable it, but I agree this change should bump version or wait 7.0.

**Comment by evcohen** [comment]
> Shall we also document in https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md that `label-has-associated-control` is designed replacement of this rule? I didn't know this until I look into this pull request.

Yes, we should! @franklinyu feel free to post a PR

**Comment by jessebeach** [comment]
@spyl94, thank you for the PR. We'll go with the documentation notice in #493 at this time. We'll remove the config line as you proposed here in the next major release.

**Pull request closed without merging**